pydm/data_plugins/plugin.py

```diff
@@ -91,25 +91,46 @@ def remove_listener(self, channel):
                 pass
 
         if channel.severity_slot is not None:
-            self.new_severity_signal.disconnect(channel.severity_slot)
+            try:
```

Comment:
Could we have a single disconnect() function that takes a signal and a slot? Then, we'll have exception handling in just one place.